tests/ci/check_so_interface.py

```diff
@@ -0,0 +1,352 @@
+set_amplitude_scale_array #smurf_command
+
+TES/FLUX RAMP FUNCTIONS
+set_tes_bias_bipolar_array #smurf_util
+set_tes_bias_high_current #smurf_util
+set_tes_bias_low_current #smurf_util
+set_mode_dc #smurf_util
+set_mode_ac #smurf_util
+flux_ramp_setup #smurf_tune.py
+
+DATA ACQUISITION FUNCTIONS
+set_stream_enable #smurf_command
+take_stream_data #smurf_util
+take_noise_psd #smurf_noise
+stream_data_on #smurf_util
+stream_data_off #surf_util
+read_stream_data #smurf_util
+set_downsample_filter #smurf_util
+
+IV FUNCTIONS
+run_iv #smurf_iv
+analyze_iv #smurf_iv
+
+DATA OUTPUTS TO DISK
+tune files generated when new resonators are found
+channel mapping file format
+.dat noise files - generated by take_stream_data
+iv_files - generated by run_iv"
+"""
+def compare_args(node, fname, classname, funcname, type):
+    # Check function
+    if type == "isfunc":
+        spec_args = interface[fname][classname][funcname]["args"]
+        spec_defaults = interface[fname][classname][funcname]["defaults"]
+
+        # Extract function arguments from the AST node
+        found_args = [arg.arg for arg in node.args.args]
+        found_defaults = [ast.unparse(d) if d else None for d in node.args.defaults]
+
+        # Compare arguments to specified arguments in interface dict
+        if found_args != spec_args:
+            print(f"Mismatch in arguments for function {funcname}")
+            print(f"Found: {found_args}")
+            print(f"Expected: {spec_args}")
+            return False
+        
+        # Compare default values to specified defaults in interface dict
+        if found_defaults != spec_defaults:
+            print(f"Mismatch in default values for function {funcname}")
+            print(f"Found: {found_defaults}")
+            print(f"Expected: {spec_defaults}")
+            return False
+    # Check class
+    elif type == "isclass":
+        spec_meths = interface[fname][classname]["methods"]
+
+        # Extract class methods from the AST node
+        found_meths = []
+        for child in node.body:
+            if isinstance(child, ast.FunctionDef):
+                found_meths.append(child.name)
+
+        # Compare methods to specified methods in interface dict
+        if found_meths != spec_meths:
+            print(f"Mismatch in methods for class {classname}")
+            print(f"Found: {found_meths}")
+            print(f"Expected: {spec_meths}")
+            return False
+    return True
+
+if __name__ == "__main__":   #if this is the main thing being run
+    for fname, classname in interface.items():   #loop over file names "fname" as keys and "spec" as values in dictionary called "interface"
+        for classname, funcname in classname.items():
+            for funcname in funcname.keys():
+                with open(fname, 'r') as fh:   #opens file "fname", in read mode 'r', to be used in code as "fh"
+                    tree = ast.parse(fh.read()) #parsing contents of file into abstract syntax tree
```

Comment:
might make sense to open the file and parse the tree in the outer loop to avoid redoing it on every iteration needlessly.